src/main/scala/atom/languageclient/ServerManager.scala

```diff
@@ -2,7 +2,116 @@ package laughedelic.atom.languageclient
 
 import io.scalajs.nodejs.{ child_process => cp }
 import laughedelic.atom.CompositeDisposable
-import scala.scalajs.js
+import scala.scalajs.js, js.|
+
+// Completion options.
+@js.native
+trait CompletionOptions extends js.Object {
+  // The server provides support to resolve additional information for a completion item.
+  val resolveProvider: js.UndefOr[Boolean] = js.native
+  // The characters that trigger completion automatically.
+  val triggerCharacters: js.UndefOr[js.Array[String]] = js.native
+}
+
+// Signature Help options.
+@js.native
+trait SignatureHelpOptions extends js.Object {
+  // The characters that trigger signature help automatically.
+  val triggerCharacters: js.UndefOr[js.Array[String]] = js.native
+}
+
+// Code Lens options.
+@js.native
+trait CodeLensOptions extends js.Object {
+  // Code lens has a resolve provider as well.
+  val resolveProvider: js.UndefOr[Boolean] = js.native
+}
+
+// Document Link options
+@js.native
+trait DocumentLinkOptions extends js.Object {
+  // Document links have a resolve provider as well.
+  val resolveProvider: js.UndefOr[Boolean] = js.native
+}
+
+// Execute Command options.
+@js.native
+trait ExecuteCommandOptions extends js.Object {
+  // The commands to be executed on the server
+  val commands: js.UndefOr[js.Array[String]] = js.native
```

Comment:
I think this one is not optional: https://github.com/atom/atom-languageclient/blob/master/lib/languageclient.js#L788